addon/components/ce/content-editable.js

```diff
@@ -273,20 +275,22 @@ export default class ContentEditable extends Component {
 
         const htmlPaste = clipboardData.getData('text/html');
         const cleanHTML = inputParser.cleanupHTML(htmlPaste);
-        inserHtml(cleanHTML);
+        this.rawEditor.executeCommand("insert-html", cleanHTML, pasteRange);
       }
       catch(e) {
         // fall back to text pasting
         console.warn(e); //eslint-disable-line no-console
         const text = this.getClipboardContentAsText(clipboardData);
-        inserHtml(text);
+        this.rawEditor.executeCommand("insert-html", text, pasteRange);
       }
     }
 
     else {
       const text = this.getClipboardContentAsText(clipboardData);
-      inserHtml(text);
+      this.rawEditor.executeCommand("insert-html", text, pasteRange);
     }
+    this.rawEditor.selection.lastRange.collapse();
+    this.rawEditor.model.writeSelection();
     this.rawEditor.updateRichNode();
```

Comment:
I think updateRichNode can be removed here, since it will already be called when executing the command